### PR TITLE
Add extension ip4r as contrib extension

### PR DESCRIPTION
**Pull request by beeender**
- Add ip4r as a git subtree
- Modify Makefiles

## Here are some reminders before you submit the pull request
- [X] Add tests for the change
- [X] Document changes
- [ ] Communicate in the mailing list if needed
- [X] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by lij55** [comment]
it looks this PR missed the license part: https://github.com/RhodiumToad/ip4r/blob/master/LICENSE.
`Permission to use, copy, modify, and distribute this software and its
documentation for any purpose, without fee, and without a written agreement is
hereby granted, provided that the above copyright notice and this paragraph
and the following two paragraphs appear in all copies.`

File: gpcontrib/ip4r/Makefile [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+VPATH := $(dir $(firstword $(MAKEFILE_LIST)))
+endif
+endif
+
+ifndef NO_PGXS
+PG_CONFIG ?= pg_config
+PGXS = $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+else
+# TODO: contrib or gpcontrib?
```
Comment (beeender) [inline gpcontrib/ip4r/Makefile:47]: remove this TODO

**Review by beeender — COMMENTED** [review]
LGTM

**Review by Sasasu — APPROVED** [review]

**Review by yihong0618 — APPROVED** [review]
LGTM

**Review by xuebinsu — APPROVED** [review]

**Comment by beeender** [comment]
We pushed this by accident manually. See #13588 

Close this one.

**Pull request closed without merging**